src/librustc_mir/transform/simplify_try.rs

```diff
@@ -172,21 +169,38 @@ impl<'tcx> MirPass<'tcx> for SimplifyBranchSame {
                         StatementKind::InlineAsm(..) => true,
                         _ => false,
                     })
-                })
-                .peekable();
+                });
 
             // We want to `goto -> bb_first`.
-            let bb_first = iter_bbs_reachable.peek().map(|(idx, _)| *idx).unwrap_or(targets[0]);
+            let first_bb_reachable = iter_bbs_reachable.next();
 
             // All successor basic blocks should have the exact same form.
-            let all_successors_equivalent =
-                iter_bbs_reachable.map(|(_, bb)| bb).tuple_windows().all(|(bb_l, bb_r)| {
-                    bb_l.is_cleanup == bb_r.is_cleanup
-                        && bb_l.terminator().kind == bb_r.terminator().kind
-                        && bb_l.statements.iter().eq_by(&bb_r.statements, |x, y| x.kind == y.kind)
-                });
+            let all_successors_equivalent = if let Some((_, mut bb_l)) = first_bb_reachable {
```

Comment:
I can't say I'm a fan of this change. Sure, itertools takes some compile time, but I think we should use it more, to be able to write nice declarative code like before this diff. I'm sure there are quite a few places in the compiler where we could clean things up with a more active "do use itertools" policy.

Reply:
I can support that I was only feeling that it was a lot of time for use of one function.
also now for some reason itertools is built twice think that is solved even with this file reverted.

Reply:
Yeah I get that. :) I'd be happy to r+ the rest of the PR though for now.

Reply:
fixed the comment now
feels like window for iterators can be in std as there is an version for slices.